EIPS/eip-add-did-related-methods.md

```diff
@@ -0,0 +1,179 @@
+
+## Specification
+Five new JSON-RPC methods are specified under the new `did_*` prefix.
+
+### Auth
+
+Authenticate the current rpc connection to the DID methods.
+
+Prompt the user to give permission to the current connection to access the user DID and the given `paths`.
+
+##### Method: 
+
+`did_authenticate`
+
+##### Params:
+
+* `paths` - an array of strings
+
+##### Returns:
+
+* `did` - the DID which authentication was given for
+* `paths` - the paths which was given permission for
+
+
+
+#### CreateJWS
+
+Creates a JSON Web Signature.
+
+An additional property `kid` with the value which represents the DID and the key that was used to sign the JWS ([details](https://github.com/decentralized-identity/did-jose-extensions/issues/2)).
+
+##### Method: 
+
+`did_createJWS`
+
+##### Params:
+
+* `payload` - the payload to sign, json object
+* `protected` - the protoected header, json object
+* `did` - the DID that should sign the message, may include the key fragment, string
+
+##### Returns:
+
+A JWS with compact serialization, string
+
+##### Recommendation:
+
+Use `secp256k1` for singning, alternatively `ed25519`.
```

Comment:
This seems like it needs to be well standardized as the recipient of the signature needs to know whether it was signed with `secp256k1` or `ed25519`.  Am I incorrect in this?  Can it be deduced which way it was signed without having to probe?

Reply:
The JWS envelope should always contain an `alg` property in the header which specifies which signature algorithm was used.